### ISSUE TITLE
Fails to create env with OpenVino 

### ISSUE THREAD

**Issue by tfmoraes** [nix-community/poetry2nix#308]
If `pyproject.toml` lists openvino as dependecy, poetry2nix is not able to create a env shell. Like this `pyproject.toml` file:

```toml
[tool.poetry]
name = "tmp"
version = "0.1.0"
description = ""
authors = ["Thiago Franco de Moraes <totonixsame@gmail.com>"]

[tool.poetry.dependencies]
python = "^3.8"
openvino = "^2021.3.0"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

And its `poetry.lock`:

```toml
[[package]]
name = "numpy"
version = "1.20.3"
description = "NumPy is the fundamental package for array computing with Python."
category = "main"
optional = false
python-versions = ">=3.7"

[[package]]
name = "openvino"
version = "2021.3.0"
description = "Inference Engine Python* API"
category = "main"
optional = false
python-versions = "*"

[package.dependencies]
numpy = ">=1.16.3"

[metadata]
lock-version = "1.1"
python-versions = "^3.8"
content-hash = "01d75b86182dbe6f2fd2c6dc46282c57d0dfbf09f267eef12ba92e894cbbf775"

[metadata.files]
numpy = [
    {file = "numpy-1.20.3-cp37-cp37m-macosx_10_9_x86_64.whl", hash = "sha256:70eb5808127284c4e5c9e836208e09d685a7978b6a216db85960b1a112eeace8"},
    {file = "numpy-1.20.3-cp37-cp37m-manylinux_2_12_i686.manylinux2010_i686.whl", hash = "sha256:6ca2b85a5997dabc38301a22ee43c82adcb53ff660b89ee88dded6b33687e1d8"},
    {file = "numpy-1.20.3-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl", hash = "sha256:c5bf0e132acf7557fc9bb8ded8b53bbbbea8892f3c9a1738205878ca9434206a"},
    {file = "numpy-1.20.3-cp37-cp37m-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:db250fd3e90117e0312b611574cd1b3f78bec046783195075cbd7ba9c3d73f16"},
    {file = "numpy-1.20.3-cp37-cp37m-manylinux_2_5_i686.manylinux1_i686.whl", hash = "sha256:637d827248f447e63585ca3f4a7d2dfaa882e094df6cfa177cc9cf9cd6cdf6d2"},
    {file = "numpy-1.20.3-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl", hash = "sha256:8b7bb4b9280da3b2856cb1fc425932f46fba609819ee1c62256f61799e6a51d2"},
    {file = "numpy-1.20.3-cp37-cp37m-win32.whl", hash = "sha256:67d44acb72c31a97a3d5d33d103ab06d8ac20770e1c5ad81bdb3f0c086a56cf6"},
    {file = "numpy-1.20.3-cp37-cp37m-win_amd64.whl", hash = "sha256:43909c8bb289c382170e0282158a38cf306a8ad2ff6dfadc447e90f9961bef43"},
    {file = "numpy-1.20.3-cp38-cp38-macosx_10_9_x86_64.whl", hash = "sha256:f1452578d0516283c87608a5a5548b0cdde15b99650efdfd85182102ef7a7c17"},
    {file = "numpy-1.20.3-cp38-cp38-manylinux_2_12_i686.manylinux2010_i686.whl", hash = "sha256:6e51534e78d14b4a009a062641f465cfaba4fdcb046c3ac0b1f61dd97c861b1b"},
    {file = "numpy-1.20.3-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl", hash = "sha256:e515c9a93aebe27166ec9593411c58494fa98e5fcc219e47260d9ab8a1cc7f9f"},
    {file = "numpy-1.20.3-cp38-cp38-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:c1c09247ccea742525bdb5f4b5ceeacb34f95731647fe55774aa36557dbb5fa4"},
    {file = "numpy-1.20.3-cp38-cp38-manylinux_2_5_i686.manylinux1_i686.whl", hash = "sha256:66fbc6fed94a13b9801fb70b96ff30605ab0a123e775a5e7a26938b717c5d71a"},
    {file = "numpy-1.20.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl", hash = "sha256:ea9cff01e75a956dbee133fa8e5b68f2f92175233de2f88de3a682dd94deda65"},
    {file = "numpy-1.20.3-cp38-cp38-win32.whl", hash = "sha256:f39a995e47cb8649673cfa0579fbdd1cdd33ea497d1728a6cb194d6252268e48"},
    {file = "numpy-1.20.3-cp38-cp38-win_amd64.whl", hash = "sha256:1676b0a292dd3c99e49305a16d7a9f42a4ab60ec522eac0d3dd20cdf362ac010"},
    {file = "numpy-1.20.3-cp39-cp39-macosx_10_9_x86_64.whl", hash = "sha256:830b044f4e64a76ba71448fce6e604c0fc47a0e54d8f6467be23749ac2cbd2fb"},
    {file = "numpy-1.20.3-cp39-cp39-manylinux_2_12_i686.manylinux2010_i686.whl", hash = "sha256:55b745fca0a5ab738647d0e4db099bd0a23279c32b31a783ad2ccea729e632df"},
    {file = "numpy-1.20.3-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.whl", hash = "sha256:5d050e1e4bc9ddb8656d7b4f414557720ddcca23a5b88dd7cff65e847864c400"},
    {file = "numpy-1.20.3-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:a9c65473ebc342715cb2d7926ff1e202c26376c0dcaaee85a1fd4b8d8c1d3b2f"},
    {file = "numpy-1.20.3-cp39-cp39-win32.whl", hash = "sha256:16f221035e8bd19b9dc9a57159e38d2dd060b48e93e1d843c49cb370b0f415fd"},
    {file = "numpy-1.20.3-cp39-cp39-win_amd64.whl", hash = "sha256:6690080810f77485667bfbff4f69d717c3be25e5b11bb2073e76bb3f578d99b4"},
    {file = "numpy-1.20.3-pp37-pypy37_pp73-manylinux_2_12_x86_64.manylinux2010_x86_64.whl", hash = "sha256:4e465afc3b96dbc80cf4a5273e5e2b1e3451286361b4af70ce1adb2984d392f9"},
    {file = "numpy-1.20.3.zip", hash = "sha256:e55185e51b18d788e49fe8305fd73ef4470596b33fc2c1ceb304566b99c71a69"},
]
openvino = [
    {file = "openvino-2021.3.0-2774-cp36-cp36m-macosx_10_15_x86_64.whl", hash = "sha256:29f7a37e7bdea4b570bf7615cb69e35aa25da3dbf4ad27ea32bdfb6a775f313a"},
    {file = "openvino-2021.3.0-2774-cp36-cp36m-manylinux2014_x86_64.whl", hash = "sha256:5569d84c704dc3c84720320f85de6ba8aaf80db64ed65c4cfb426e5eb25de65f"},
    {file = "openvino-2021.3.0-2774-cp36-cp36m-win_amd64.whl", hash = "sha256:da72bc07f3d3d0d8dd44fabe0e147689f57677febef0d6f31704d58c92ecf5bc"},
    {file = "openvino-2021.3.0-2774-cp37-cp37m-macosx_10_15_x86_64.whl", hash = "sha256:1ac72d346c263d89f353a6b3a512d2538e6f75b5ca206b3f1913dc7324d4e1c9"},
    {file = "openvino-2021.3.0-2774-cp37-cp37m-manylinux2014_x86_64.whl", hash = "sha256:92ad4c8e9aea997669370b46cad2176d5b3bf805c629ab40fec8ce5e36af13d0"},
    {file = "openvino-2021.3.0-2774-cp37-cp37m-win_amd64.whl", hash = "sha256:0e4088d00fdf9c707f6bd249cebcb8f365ce60b148f4f31aed21b24548d0b43f"},
    {file = "openvino-2021.3.0-2774-cp38-cp38-macosx_10_15_x86_64.whl", hash = "sha256:b6b60ff093da9f218154ae2492766abfcf8ef41e36d01f4285d4bb34d4f6b507"},
    {file = "openvino-2021.3.0-2774-cp38-cp38-win_amd64.whl", hash = "sha256:94fabf97ef9a39f70203c38376353c4e93ec210a2faae088ab4b6290a0328bc1"},
    {file = "openvino-2021.3.0-3029-cp38-cp38-manylinux2014_x86_64.whl", hash = "sha256:e00f4764a3d3d1ed3e9ec9c06b379a024ae84156d594a078e521a8af5c062ce0"},
]
```

And this is the `shell.nix`:

```nix
{ pkgs ? import <nixpkgs> {} }:
let
  myAppEnv = pkgs.poetry2nix.mkPoetryEnv {
    projectDir = ./.;
  };
in myAppEnv.env
```

This is the error:

```shell
❯ nix-shell --show-trace                                                                                                                                                      poetry 
error: list index 0 is out of bounds

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/development/tools/poetry2nix/poetry2nix/mk-poetry-dep.nix:73:27:

           72|           entries = (if preferWheel then binaryDist ++ sourceDist else sourceDist ++ binaryDist) ++ eggs;
           73|           lockFileEntry = builtins.head entries;
             |                           ^
           74|           _isEgg = isEgg lockFileEntry;

       … while evaluating the attribute 'name'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/development/tools/poetry2nix/poetry2nix/mk-poetry-dep.nix:78:11:

           77|           inherit (lockFileEntry) file hash;
           78|           name = file;
             |           ^
           79|           format =

       … while evaluating 'hasPrefix'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/lib/strings.nix:218:5:

          217|     # Input string
          218|     str: substring 0 (stringLength pref) str == pref;
             |     ^
          219|

       … from call site

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/lib/strings.nix:258:8:

          257|       drop = x: substring 1 (stringLength x) x;
          258|     in hasPrefix infix content
             |        ^
          259|       || content != "" && hasInfix infix (drop content);

       … while evaluating 'hasInfix'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/lib/strings.nix:255:21:

          254|   */
          255|   hasInfix = infix: content:
             |                     ^
          256|     let

       … from call site

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/development/tools/poetry2nix/poetry2nix/lib.nix:79:8:

           78|     in
           79|     if lib.strings.hasInfix "manylinux1" f then { pkg = [ ml.manylinux1 ]; str = "1"; }
             |        ^
           80|     else if lib.strings.hasInfix "manylinux2010" f then { pkg = [ ml.manylinux2010 ]; str = "2010"; }

       … while evaluating 'getManyLinuxDeps'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/development/tools/poetry2nix/poetry2nix/lib.nix:75:22:

           74|   #
           75|   getManyLinuxDeps = f:
             |                      ^
           76|     let

       … from call site

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/development/tools/poetry2nix/poetry2nix/mk-poetry-dep.nix:119:38:

          118|         ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) pythonPackages.setuptools
          119|         ++ lib.optional (!isSource) (getManyLinuxDeps fileInfo.name).pkg
             |                                      ^
          120|         ++ lib.optional isLocal buildSystemPkgs

       … while evaluating anonymous lambda

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/stdenv/generic/make-derivation.nix:152:17:

          151|           (map (drv: drv.__spliced.hostHost or drv) depsHostHost)
          152|           (map (drv: drv.crossDrv or drv) buildInputs)
             |                 ^
          153|         ]

       … from call site

       … while evaluating 'getOutput'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/lib/attrsets.nix:486:23:

          485|   */
          486|   getOutput = output: pkg:
             |                       ^
          487|     if pkg.outputUnspecified or false

       … from call site

       … while evaluating the attribute 'buildInputs' of the derivation 'python3.8-openvino-2021.3.0'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/stdenv/generic/make-derivation.nix:201:11:

          200|         // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          201|           name =
             |           ^
          202|             let

       … while evaluating the attribute 'outPath'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/lib/customisation.nix:164:7:

          163|       drvPath = assert condition; drv.drvPath;
          164|       outPath = assert condition; drv.outPath;
             |       ^
          165|     };

       … while evaluating anonymous lambda

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/lib/lists.nix:645:24:

          644|    */
          645|  unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
             |                        ^
          646|

       … from call site

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/top-level/python-packages.nix:67:6:

           66|     modules = filter hasPythonModule drvs;
           67|   in unique ([python] ++ modules ++ concatLists (catAttrs "requiredPythonModules" modules));
             |      ^
           68|

       … while evaluating 'requiredPythonModules'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/top-level/python-packages.nix:65:27:

           64|   # Get list of required Python modules given a list of derivations.
           65|   requiredPythonModules = drvs: let
             |                           ^
           66|     modules = filter hasPythonModule drvs;

       … from call site

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/development/interpreters/python/wrapper.nix:20:13:

           19|   env = let
           20|     paths = requiredPythonModules (extraLibs ++ [ python ] ) ;
             |             ^
           21|     pythonPath = "${placeholder "out"}/${python.sitePackages}";

       … while evaluating the attribute 'passAsFile'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/build-support/buildenv/default.nix:77:5:

           76|     # XXX: The size is somewhat arbitrary
           77|     passAsFile = if builtins.stringLength pkgs >= 128*1024 then [ "pkgs" ] else [ ];
             |     ^
           78|   }

       … while evaluating the attribute 'passAsFile' of the derivation 'python3-3.8.9-env'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/stdenv/generic/make-derivation.nix:201:11:

          200|         // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          201|           name =
             |           ^
          202|             let

       … while evaluating the attribute 'nativeBuildInputs' of the derivation 'interactive-python3-3.8.9-environment'

       at /nix/store/2y8xq739cy0klq7mh92fxziahda70ryd-source/pkgs/stdenv/generic/make-derivation.nix:201:11:

          200|         // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          201|           name =
             |           ^
          202|             let
```

**Comment by tfmoraes** [comment]
I think this error is happening because OpenVino package has build tag in its filename, and this tag is optional https://www.python.org/dev/peps/pep-0427/#file-name-convention

**Comment by tfmoraes** [comment]
closed by #612
